airflow/providers/http/sensors/http.py

```diff
@@ -91,15 +100,26 @@ def __init__(
         self.headers = headers or {}
         self.extra_options = extra_options or {}
         self.response_check = response_check
-
-        self.hook = HttpHook(method=method, http_conn_id=http_conn_id)
```

Comment:
cc: @phanikumv

Reply:
This is the error we get in our DAG run with the RC:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/astronomer/providers/http/sensors/http.py", line 67, in execute
    method=self.hook.method,  # TODO: Fix this to directly get method from ctor
AttributeError: 'HttpSensorAsync' object has no attribute 'hook'
```

Reply:
I have created a PR to reinstantiate the hook in the derived class: https://github.com/astronomer/astronomer-providers/pull/515

Reply:
Well. I have not thought too much about this use case when I added it - it was just bringing a general approach that we usually do where Hooks are created in execute() method rather than in constructor. 

But if there are good reasons why it should not be moved I can remove that one from the batch and make RC2 and restore the self.hook. 

I think with async operators approach we miss a bit of best practices:

* should we create Hook in init()?
* should we have a Hook as Operator attribute (self.hook)?
* or should we restore the hook every time when we come back from deferred state?

I think the last option is most "reasonable" because first two might make (an incorrect) assumption that the same hook object is available during the whole lifecycle of an operator's methods, and if Hook is not "stateless", it might break when operator is deferred (because the Hook will be recreated when operator resumes from deferred state). If you keep the hook as `self.` property, you might (unconciously) rely on the fact that the hook is the same object in various methods of deferred operator.

WDYT? Does it sound like a good practice ?  Also others @dstandish maybe? I think we had some discussion about this in the past (can't recall exactly)

Reply:
Makes sense to me. In that case, I think we can create a clean up story on our end to adhere to the best practice post our conclusion on this discussion. We do not need an RC2 for that then, and also for the sake that the operator still continues to works, I will for the time being put the hack to create the hook in our operator init() with our PR. 

Also looping in @kaxil @bharanidharan14 @rajaths010494 @pankajastro for the discussion

Reply:
That sounds good to me. Would be great to contribute such "best practicess for writing Defferable Operators" to the community after all the experiences Astronomer had with writing theirs. It would be a great contribution :).

Later on we could event try to automate some of those "best practices" in pre-commits for any future community async operators. This sounds like a doable task to ast-parse an operatort and if it is a Defferrable one, make sure no self.hook is used.